### PR TITLE
ziggurat: change and make more flexible scrying pyro chain state

### PR DESCRIPTION
**Pull request by hosted-fornet**
**Problem**:

1. Chain state scrying only gives state.
2. Can't manipulate that data (which is often big). 

**Solution**:

1. Scry now returns tx.s as well. 
2. Add poke similar to %pyro-agent-state with `grab` field.

**Notes**:

FYI @0x70b1a5 see new interface

~~WIP~~

### CODE REVIEW

File: lib/zig/ziggurat.hoon [hunk 1]
```diff
@@ -2482,6 +2489,7 @@
         [%send-pyro-dojo (ot ~[[%who (se %p)] [%command sa]])]
     ::
         [%pyro-agent-state pyro-agent-state]
+        [%pyro-chain-state pyro-chain-state]
     ::
         [%change-focus ul]
         [%add-project-link ul]
```
Comment (0x70b1a5) [inline lib/zig/ziggurat.hoon:2495]: on line 2506 below be sure to change it to 
```
:^    [%test-result-num-characters ni]
    [%state-num-characters ni]
  [%compiler-error-num-lines ni]
~
```

**Review by 0x70b1a5 — CHANGES_REQUESTED** [review]

File: lib/zig/ziggurat.hoon (hunk 1 above)
Comment (hosted-fornet) [reply lib/zig/ziggurat.hoon:2495]: Fixed in e072c2d

**Comment by hosted-fornet** [comment]
@0x70b1a5 I have been debating whether to add a `town-id=@ux` to %pyro-chain-state so that it returns a `batch:ui` rather than a `(map @ux batch:ui)`. Let me know if one or the other would work better for you. Otherwise, this PR is done.

**Comment by hosted-fornet** [comment]
To make this practical, I need to add imports (and probably makes sense to add to %pyro-agent-state as well -- sorry for breaking changes). Reason it is required here is that we use `merk` maps rather than stdlib maps, which means accessing elements of the maps doesn't work properly with, e.g., `get:by` since map order is not as `by` expects.

**Comment by hosted-fornet** [comment]
Ok, finished adding `test-imports` field to %pyro-agent-state and %pyro-chain-state in 89c68a4.

**Pull request merged**